docs/DeveloperGuide.md

```diff
@@ -270,64 +271,83 @@ shows the updated Sequence diagram for the executing of our `delete` command.
+#### Resident Information
 
-In `RC4HDB`, we reworked the entire configuration for displaying results. We replaced `PersonListPanel` and `PersonCard`
-by a `ResidentTableView` component.
+The `ResidentTabView` contains a `ResidentTableView` which is implemented via the `TableView` class of `JavaFX`. This is represented
+as a table, where each row corresponds to a `Resident` in `RC4HDB`, and each column corresponds to a field belonging to that `Resident`.
 
-<img src="images/UiClassDiagram.png" width="550" />
+##### Design considerations
 
-`ResidentTableView` is implemented via the `TableView` class of `JavaFX`. Graphically, `ResidentTableView` is presented
-as a table. Each row corresponds to a `Resident` in `RC4HDB`, and each column corresponds to a field belonging to that `Resident`.
+Aspect: Display format
 
-From the users' viewpoint, the data generated in `ResidentTableView` is a single unit, but it is logically separated
-into two distinct parts. The first part being the first column which is the `IndexColumn` and the second being all other
-columns, also known as `FieldColumns`.
+Alternative 1 (current choice): Table
 
-The main reason for this distinction is *method to generate the cell values*.
-- The indices in the `IndexColumn` are generated independently to the `FieldColumns`. This is because fields within
-  `Resident` do not affect its index within the table. In two different commands, the same `Resident` could
-  have different indices in the results.
-- In contrast, in the generation of values for each cell in `FieldColumn`, values are obtained by iterating
-through a list of `Residents` and setting each cell to it. As the iterator does not modify the ordering of `Residents`,
-  the same technique is applied to obtain the values for other `FieldColumns`.
+Pros:
+* Ability to display condensed information clearly
+* Ability to manipulate the data that can be displayed by showing and hiding field columns
+* Ability for the user to view large amounts of information at a glance
+* Powerful TableView implementation allows us to dynamically obtain resident field, and update the table when the Model changes
 
-As a consequence, fields of a `Resident` will always collectively be together in the same row, though it may appear in
-two different indices in the results of two different commands.
+Cons:
+* Possible information overload on the user
+* Possible performance issues in terms of memory usage when the number of columns exceed 60
 
-<br>
+For more information on how the TableView implementation is powerful, we refer you to the [documentation](https://docs.oracle.com/javase/8/javafx/api/javafx/scene/control/cell/PropertyValueFactory.html)
+on *method matching*, which is what we heavily used to fetch resident data.
 
-##### Obtaining `Resident` fields
+For more information on possible performance issue, we refer you to the GitHub issue [here](https://github.com/javafxports/openjdk-jfx/issues/409).
```

Comment:
"For more information on possible performance **issues**, refer to this GitHub issue..."